src/test/java/teammates/sqlui/webapi/CreateNotificationActionTest.java

```diff
@@ -0,0 +1,166 @@
+package teammates.sqlui.webapi;
+
+import java.util.UUID;
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.NotificationStyle;
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.util.Const;
+import teammates.storage.sqlentity.Notification;
+import teammates.ui.output.NotificationData;
+import teammates.ui.request.InvalidHttpRequestBodyException;
+import teammates.ui.request.NotificationCreateRequest;
+import teammates.ui.webapi.CreateNotificationAction;
+
+/**
+ * SUT: {@link CreateNotificationAction}.
+ */
+public class CreateNotificationActionTest extends BaseActionTest<CreateNotificationAction> {
+    private static final String GOOGLE_ID = "user-googleId";
+    private static final String INVALID_TITLE = "";
+    NotificationCreateRequest testReq;
+    private final Notification testNotification = getTypicalNotificationWithId();
+
+    @Override
+    String getActionUri() {
+        return Const.ResourceURIs.NOTIFICATION;
+    }
+
+    @Override
+    String getRequestMethod() {
+        return POST;
+    }
+
+    @BeforeMethod
+    void setUp() {
+        loginAsAdmin();
+    }
+
+    @Test(enabled = false)
+    void testExecute_addNotification_success() throws Exception {
+        long startTime = testNotification.getStartTime().toEpochMilli();
+        long endTime = testNotification.getEndTime().toEpochMilli();
+        NotificationStyle style = testNotification.getStyle();
+        NotificationTargetUser targetUser = testNotification.getTargetUser();
+        String title = testNotification.getTitle();
+        String message = testNotification.getMessage();
+
+        NotificationCreateRequest req = getTypicalCreateRequest();
+        CreateNotificationAction action = getAction(req);
+        NotificationData res = (NotificationData) action.execute().getOutput();
+
+        Notification createdNotification = mockLogic.getNotification(UUID.fromString(res.getNotificationId()));
```

Comment:
Maybe consider something like this for the test
```suggestion
        when(mockLogic.createNotification(isA(Notification.class))).thenAnswer(invocation -> invocation.getArgument(0));
        NotificationCreateRequest req = getTypicalCreateRequest();
        CreateNotificationAction action = getAction(req);
        NotificationData res = (NotificationData) action.execute().getOutput();

        Notification createdNotification = testNotification;
```